formatting/src/main/java/com/avast/metrics/dropwizard/formatting/Formatter.java

```diff
@@ -0,0 +1,16 @@
+package com.avast.metrics.dropwizard.formatting;
+
+/**
+ * Generic formatter of metric names and values.
+ */
+public interface Formatter {
+    /**
+     * Separator of name parts.
+     */
+    String nameSeparator();
+
+    /**
+     * Sanitize part of a metric name to not contain any illegal character. Escape, replace or delete them.
+     */
+    String sanitizeName(String namePart);
```

Comment:
Sanitize. For example '.' is separator in graphite and it's internally mapped to filesystem directories. '/' is directory separator so it's invalid too. There are more of them and they are impossible to escape.